File: backend/backend/src/main/java/com/activitytracker/backend/dto/ActivityDto.java
package com.activitytracker.backend.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;

import java.time.LocalDate;
import java.time.OffsetDateTime;
import java.util.UUID;

public record ActivityDto(
        @NotBlank
        String activityName,

        @NotNull
        LocalDate activityDate,

        @NotNull
        UUID userId,

        @NotNull
        OffsetDateTime createdAt) {
}

File: android/app/src/main/java/com/example/activitytracker/data/remote/dto/ActivityDto.kt
package com.example.activitytracker.data.remote.dto

import com.example.activitytracker.data.local.entity.ActivityEntity
import java.util.UUID

data class ActivityUploadDto(
    val activityName: String,
    val activityDate: String,
    val createdAt: String,
    val userId: UUID
)

fun ActivityEntity.toUploadDto(userId: UUID) = ActivityUploadDto(
    activityName = activityName,
    activityDate = activityDate.toString(),
    createdAt = createdAt.toString(),
    userId = userId
)